[JuliaPoo] iP

JuliaPoo opened the pull request:
# Meow

> Meow meow says meow meow ~ Cat ([source](https://en.wikipedia.org/wiki/Meow))

Meow meow

- Meow
- Meow
- ~~Woof~~ Meow

Meow
1. Meow
2. Meow
3. Meow

**MEOW**

- [ ] Meow
- [ ] Meow
- [x] Meow

```java
public class Meow {
    public static void meow(String[] kittens) {
        Stream.of(kittens).forEach(k -> k.meow());
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

JuliaPoo committed:
Feat: Completed level 0

JuliaPoo committed:
Feat: Completed level 1

JuliaPoo committed:
Feat: Completed level 2

JuliaPoo committed:
Feat: Completed level 3

JuliaPoo committed:
Feat: Completed level 4

JuliaPoo committed:
Feat: Completed level 5

JuliaPoo committed:
Feat: Completed level 6

JuliaPoo committed:
Refactor: Use enums for task id

JuliaPoo committed:
Refactor: Follow java conventions

JuliaPoo committed:
Feat: Completed level 7

JuliaPoo committed:
Merge pull request #1 from JuliaPoo/branch-Level-7

Feat: Completed level 7

JuliaPoo committed:
Feat: Completed level 8

JuliaPoo committed:
Merge pull request #2 from JuliaPoo/branch-Level-8

Feat: Completed level 8

JuliaPoo committed:
Refactor: Completed A-MoreOOP

JuliaPoo committed:
Refactor: Completed A-Packages

JuliaPoo committed:
Merge pull request #3 from JuliaPoo/add-gradle-support

Add gradle support

JuliaPoo committed:
Build: Completed A-Gradle

JuliaPoo committed:
Refactor: Split classes up into files

In preparation for testing with JUnit

JuliaPoo committed:
Test: Completed A-JUnit

JuliaPoo committed:
Build: Completed A-Jar

JuliaPoo committed:
Refactor: Completed A-CodingStandard

JuliaPoo committed:
Doc: Completed A-JavaDoc

JuliaPoo committed:
Feat: Completed Level-9

JuliaPoo committed:
Merge pull request #6 from JuliaPoo/branch-Level-9

Feat: Completed Level-9

JuliaPoo committed:
Merge pull request #4 from JuliaPoo/branch-A-CodingStandard

Refactor: Completed A-CodingStandard

JuliaPoo committed:
Merge branch 'master' into branch-A-JavaDoc

JuliaPoo committed:
Merge pull request #5 from JuliaPoo/branch-A-JavaDoc

Doc: Completed A-JavaDoc

ziiqii commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,63 @@
+   */
+  public void run() {
+    Ui.greet();
+    boolean b = true;
```
Maybe could rename boolean b to something more descriptive like isBye to follow the coding standard?

ziiqii commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,145 @@
+    String[] cmdSplit = cmdString.split(" ");
+    String command = cmdSplit[0];
+    switch (command) {
+      case "end":
```
Could align 'case' to 'switch' here to conform to the coding standards.

ziiqii commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,44 @@
+    return id.toString();
+  }
+
+  public String toString() {
```
Good practice to include @override for toString() method

ziiqii commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,155 @@
+    return 0 <= idx && idx < storedTasks.size();
+  }
+
+  public int numberOfTask() {
```
Maybe can change to getNumberOfTasks (start with verb) to follow the coding style

ziiqii commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,174 @@
+    scanner = new Scanner(System.in);
+  }
+
+  static void inputPrompt() {
```
Maybe could use verb in method name like printInputPrompt

ziiqii reviewed (commented):
Overall, your code is neat, and follows the style guide. LGTM!

JuliaPoo committed:
Style: Used Checkstyle to reformat code

JuliaPoo committed:
Merge pull request #7 from JuliaPoo/branch-A-CheckStyle

Style: Used Checkstyle to reformat code

JuliaPoo committed:
Style: Reformatted more code

JuliaPoo committed:
Feat: Completed level 10

JuliaPoo committed:
Merge pull request #8 from JuliaPoo/branch-Level-10

Branch level 10

JuliaPoo committed:
Refactor: Minor refactor and added assertions

Due to not typing the different commands,
length of arguments in UI class is assumed
correct, which assumes Parser is implemented
correctly.

Added assertions in UI class to ensure correctness.

JuliaPoo committed:
Merge pull request #9 from JuliaPoo/branch-A-Assertions

Refactor: Minor refactor and added assertions

JuliaPoo committed:
Refactor: Seperated Ui into Command interfaces

Ui class contained all functionality of Commands,
which is messy since Ui is also modifying the TaskList.

Handling of commands is refactored into private Command
classes abstracted by CommandFactory.

This not only improves SLAP, but also removes the length
checks present in Ui previously, due to better typing.

JuliaPoo committed:
Merge pull request #10 from JuliaPoo/branch-A-CodeQuality

Refactor: Seperated Ui into Command interfaces

JuliaPoo committed:
Refactor; Moved Task creation into TaskFactory

Improves SLAP so now TaskList is no longer responsible
for creating Task, and we can move the deserealising
of Task objects from Storage into TaskFactory.

JuliaPoo committed:
Feat: Completed B-FixedDurationTasks

JuliaPoo committed:
Merge pull request #11 from JuliaPoo/branch-B-FixedDurationTasks

Feat: Completed B-FixedDurationTasks

JuliaPoo committed:
Refactor: Changed "duke" to "anna"

JuliaPoo committed:
Fix: Used == when it should be using object compare

JuliaPoo committed:
Feat: Completed A-UserGuide

JuliaPoo committed:
Merge pull request #13 from JuliaPoo/branch-A-UserGuide

Feat: Completed A-UserGuide

JuliaPoo committed:
Test: Fixed broken test in ParserTest.java

JuliaPoo committed:
Addressed Ip evaluators

JuliaPoo committed:
Addressed automatic evaluators
